packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,119 @@
+import { Injectable, Scope } from 'graphql-modules';
+import { z } from 'zod';
+import { QueryAuditLogsArgs } from '../../../__generated__/types.next';
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { Logger } from '../../shared/providers/logger';
+import { AuditLogEvent, auditLogSchema } from './audit-logs-types';
+
+export const auditLogDbObject = z.object({
+  id: z.string(),
+  event_time: z.string(),
+  user_id: z.string(),
+  user_email: z.string(),
+  organization_id: z.string(),
+  event_action: z.string(),
+  metadata: z.string().transform(x => JSON.parse(x)),
+});
+
+export type AuditLogModel = z.infer<typeof auditLogDbObject>;
+
+const auditLogCount = z.object({
+  'COUNT()': z.string(),
+});
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+export class AuditLogManager {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogsManager' });
+  }
+
+  async createLogAuditEvent(event: AuditLogEvent): Promise<void> {
+    try {
+      const { eventType, organizationId, user } = event;
+      this.logger.info('Creating a log audit event (event=%o)', event);
+
+      const parsedEvent = auditLogSchema.parse(event);
+      const metadata = {
+        user: user.user,
+        ...parsedEvent,
+      };
+
+      const eventMetadata = JSON.stringify(metadata);
+      const eventTime = new Date();
+
+      const values = [
+        eventTime,
+        user.userId,
+        user.userEmail,
+        organizationId,
+        eventType,
+        eventMetadata,
+      ];
+
+      await this.clickHouse.insert({
+        query: sql`
+        INSERT INTO audit_log
+        (event_time, user_id, user_email, organization_id, event_action, metadata)
+        FORMAT CSV`,
+        data: [values],
+        timeout: 5000,
+        queryId: 'create-audit-log',
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  async getPaginatedAuditLogs(props: QueryAuditLogsArgs): Promise<AuditLogModel[]> {
+    this.logger.info(
+      'Getting paginated audit logs (limit=%s, offset=%s)',
+      props.limit,
+      props.offset,
+    );
+
+    const limit = props.limit ?? 25; // Default to 25 if limit is undefined
+    const sqlLimit = sql.raw(limit.toString());
```

Comment:
I think you need to use something else than `sql.raw` here, to make it safer? @kamilkisiela @n1ru4l